From f1243236e46e41696216776b9e7774f4a79a7a43 Mon Sep 17 00:00:00 2001
From: Eric Sauvageau <merlin@asuswrt-merlin.net>
Date: Thu, 17 Sep 2026 17:50:30 -0400
Subject: [PATCH] webui: when disabling auto-refresh on System Log, also
 disable querying the router

(cherry picked from commit 0c5529942a71998e8bc0d333b65ad00fb7676aa3)
---
 release/src/router/www/Main_LogStatus_Content.asp | 14 ++++++++++----
 1 file changed, 10 insertions(+), 4 deletions(-)

diff --git a/release/src/router/www/Main_LogStatus_Content.asp b/release/src/router/www/Main_LogStatus_Content.asp
index 00139ceed3..d47dec8355 100644
--- a/release/src/router/www/Main_LogStatus_Content.asp
+++ b/release/src/router/www/Main_LogStatus_Content.asp
@@ -137,17 +137,23 @@ $.getJSON("/ajax/logFilter.json", function(data){
 var height = 0;
 function get_log_data(){
 	var h = 0;
-    $.ajax({
+
+	if (!document.getElementById("auto_refresh").checked) {
+		setTimeout(get_log_data, 3000)
+		return;
+	}
+
+	$.ajax({
 		url: '/appGet.cgi?hook=nvram_dump(\"syslog.log\",\"syslog.sh\")',
 		dataType: 'text',
 		error: function(xhr){
-      		setTimeout(get_log_data, 1000);
+			setTimeout(get_log_data, 1000);
 		},
 		success: function(response){
 			var logString = htmlEnDeCode.htmlEncode(response.toString().slice(26,-4));
-    		h = $("#textarea").scrollTop();
+			h = $("#textarea").scrollTop();
 			var _log = '';
-			if((document.getElementById("auto_refresh").checked) && !(height > 0 && h < height)){
+			if(!(height > 0 && h < height)){
 				var _string = logString.split('\n');
 				for(var i=0;i<_string.length;i++){
 					var found = filter.find(function(e){
-- 
2.25.1

